src/oemof/visio/energy_system_graph.py

```diff
@@ -161,68 +195,118 @@ def __init__(
             else:
                 img_format = "pdf"
 
-        self.dot = graphviz.Digraph(format=img_format, **kwargs)
+        self.legend = legend
+        self.img_format = img_format
+        self.digraph_kwargs = kwargs
         self.txt_width = txt_width
         self.txt_fontsize = str(txt_fontsize)
-        self.busses = []
 
-        if legend is True:
-            with self.dot.subgraph(name="cluster_1") as c:
-                # color of the legend box
-                c.attr(color="black")
-                # title of the legend box
-                c.attr(label="Legends")
-                self.add_bus(subgraph=c)
-                self.add_sink(subgraph=c)
-                self.add_source(subgraph=c)
-                self.add_transformer(subgraph=c)
-                self.add_storage(subgraph=c)
+    def add_components(self, components, subgraph=None, depth=0):
+        if SUBNETWORK_MODULE is True:
+            subnetworks = [
+                n for n in components if n.depth == depth and isinstance(n, SubNetwork)
+            ]
+            atomicnodes = [
+                n
+                for n in components
+                if n.depth == depth and not isinstance(n, SubNetwork)
+            ]
 
-        # draw a node for each of the energy_system's component.
-        # the shape depends on the component's type.
-        for nd in energy_system.nodes:
+            # draw the subnetworks recursively
+            if subnetworks:
+                for sn in subnetworks:
+                    self.add_subnetwork(sn, subgraph=subgraph, depth=depth)
+
+        else:
+            atomicnodes = components
+
+        if atomicnodes:
+            components_to_add = atomicnodes
+
+        busses = []
+
+        for nd in components_to_add:
+            # Back-compatibility with oemof-solph 0.6.0 and lower
+            if SUBNETWORK_MODULE is False:
+                if not hasattr(nd, "depth"):
+                    setattr(nd, "depth", 0)
```

Comment:
In my opinion, `visio` should not alter the data that it means to display.

Reply:
I agree we can just requier the latest solph version

Reply:
Removed in f726c9aed629d53a0cc58afd7667b83cf3a9176f